tests/nativeapp/test_run_processor.py

```diff
@@ -90,12 +91,11 @@ def _get_na_run_processor():
         project_root=dm.project_root,
     )
 
-
-# Test create_dev_app with exception thrown trying to use the warehouse
+# Test create_dev_app with no existing application AND create succeeds AND app role == package role
+@mock.patch(RUN_PROCESSOR_GET_EXISTING_APP_INFO, return_value=None)
 @mock.patch(NATIVEAPP_MANAGER_EXECUTE)
-@mock_connection()
-def test_create_dev_app_w_warehouse_access_exception(
-    mock_conn, mock_execute, temp_dir, mock_cursor
+def test_use_role_in_create_dev_app_w_no_additional_privileges(
+    mock_execute, mock_get_existing_app_info, temp_dir, mock_cursor
```

Comment:
this `test_use_role_in_create_dev_app_w_no_additional_privileges` and the below `test_use_role_in_create_dev_app_w_additional_privileges ` are the two that test use_role flows in `create_or_upgrade`.